src/features/profile/Profile.tsx

```diff
@@ -0,0 +1,186 @@
+  username: string;
+  viewMine: boolean;
+  viewNarratives: boolean;
+}
+
+export const Profile: FC<ProfileParams> = ({
+  narrativesLink,
+  pageTitle,
+  profileLink,
+  realname,
+  username,
+  viewMine,
+  viewNarratives,
+}) => {
+  usePageTitle(pageTitle);
+  return (
+    <>
+      <nav className={classes['profile-nav']}>
+        <ul className="links">
+          <li>
+            <Link to={profileLink}>Profile</Link>
+          </li>
+          <li>
+            <Link to={narrativesLink}>Narratives</Link>
+          </li>
+        </ul>
+      </nav>
+      <section className={classes['profile-narratives']}>
+        {viewNarratives ? (
+          <NarrativesView realname={realname} yours={viewMine} />
+        ) : (
+          <ProfileView realname={realname} />
+        )}
+        <Button>Edit</Button>
+      </section>
+    </>
+  );
+};
+
+export const ProfileWrapper: FC = () => {
+  const dispatch = useAppDispatch();
+  const cachedProfiles = useAppSelector(cache);
+  const errorMessage = useAppSelector(error);
+  const token = useAppSelector((state) => state.auth.token);
+  const usernameAuthed = useAppSelector(authUsername);
+  const tokenString = token ? token : '';
+  const location = useLocation();
+  const viewURL = location.pathname.split('/').slice(-1)[0];
+  const viewNarratives = viewURL === 'narratives';
+  const { usernameRequested } = useParams<{ usernameRequested: string }>();
+  // Was a username specified in the URL?
+  const usernameNotSpecified =
+    usernameRequested === 'narratives' || usernameRequested === undefined;
+  // Am I looking at my own profile?
+  const viewMine = usernameRequested === usernameAuthed || usernameNotSpecified;
+  // In any case, whose profile should be shown?
+  const viewUsername =
+    usernameAuthed && usernameNotSpecified ? usernameAuthed : usernameRequested;
+  const usernameSelectedState = useAppSelector(usernameSelected);
+  // If the username has not yet been selected.
+  if (usernameSelectedState !== viewUsername) {
+    dispatch(selectUsername(viewUsername));
+    return <>Loading</>;
+  }
+  // If there is an error, for now display Page Not Found.
+  if (errorMessage) {
+    // eslint-disable-next-line no-console
+    console.error(`Error message: ${errorMessage}`);
+    dispatch(clearError());
+    return <PageNotFound />;
+  }
+  if (!usernameAuthed) {
+    /* If this component is loaded first (eg. a full page refresh) then the
+        authentication state will need to be populated before displaying the
+        profile for the current user.
+    */
+    return <>Loading authentication state.</>;
+  }
+  const loadUsername = usernameNotSpecified ? usernameAuthed : viewUsername;
+  if (!(loadUsername in cachedProfiles)) {
+    dispatch(loadProfile({ token: tokenString, username: loadUsername }));
+  }
+  const profile = cachedProfiles[loadUsername];
+  if (!profile) {
+    return <>Loading user profile.</>;
+  }
+  const profileNames = cachedProfiles[loadUsername].user;
```

Comment:
This is pretty small, but can't `cachedProfiles[loadUsername].user` just be `profile.user` since you checked that it's defined?

Reply:
Updated in the latest commit.